include/aspect/newton.h

```diff
@@ -247,6 +247,11 @@ namespace aspect
         void
         execute (internal::Assembly::Scratch::ScratchBase<dim>  &scratch_base,
                  internal::Assembly::CopyData::CopyDataBase<dim> &data_base) const override;
+
+        /**
+         * Create AdditionalMaterialOutputsStokesRHS if we need to do so.
```

Comment:
The documentation should probably actually be modified a bit further now that the derivates are handled with the same function. How does the following sound?:
` Create additional material models outputs for assembly of derivatives or adding additional terms to the right hand side of the Stokes equations. The latter could include viscoelastic forces or other user-defined values calculated within the material model`

Reply:
looks good to me. Because it look kind of weird to me that it is in the `NewtonStokesIncompressibleTerms` class, while the derivative are in principle also for the compressible case, I looked around a bit more in the code. 

The central part of this code is in the `NewtonHandler` `set_assemblers` function in `newton.cc` line 51. There it always inserts the `NewtonStokesIncompressibleTerms` assembler. The reason the test failed is because before you created the specific `create_additional_material_model_outputs` function, it used the default implantation of the `NewtonInterface`, and now that is overwritten. 

Since the `NewtonStokesIncompressibleTerms` is always used, even in the compressible case, I do not see a problem keeping it here.